charms/kfp-api/metadata.yaml

```diff
@@ -0,0 +1,34 @@
+name: kfp-api
+display-name: Kubeflow Pipelines API
+summary: Reusable end-to-end ML workflows built using the Kubeflow Pipelines SDK
+description: |
+  Machine learning (ML) toolkit that is dedicated to making deployments
+  of ML workflows on Kubernetes simple, portable, and scalable.
+maintainers:
+  - "Dominik Fleischmann <dominik.fleischmann@canonical.com>"
+  - "Kenneth Koski <kenneth.koski@canonical.com>"
+tags: [ai, big-data, kubeflow, machine-learning, tensorflow]
+series: [kubernetes]
+resources:
+  oci-image:
+    type: oci-image
+    description: Backing OCI image
+    auto-fetch: true
+    upstream-source: gcr.io/ml-pipeline/api-server:1.5.0-rc.0
+requires:
+  mysql:
+    interface: mysql
+  object-storage:
+    interface: object-storage
+    schema: https://raw.githubusercontent.com/canonical/operator-schemas/master/object-storage.yaml
+    versions: [v1]
+  kfp-viz:
+    interface: k8s-service
+    schema: https://raw.githubusercontent.com/canonical/operator-schemas/master/k8s-service.yaml
```

Comment:
Why is service-port type `string`

Reply:
The advice I've seen that generally works well is that you should only use integers for things that you use math on. Between that and a string being required in a situation like `args: ['--port', config['port']]` (because the pod spec specifies a list of strings only, and doesn't try to convert the list), I chose a string type.

That being said, elsewhere the pod spec requires ports to be specified as integers, such as in the `ports: {..}` section. So if the first paragraph didn't convince you that it was worth using a string, I don't have any qualms about switching it.

Reply:
I think we can leave it like that, not a hill I need to die on. I just wanted to make sure it was intentional.